test-loop-tests/src/setup/state.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::atomic::AtomicBool;
+use std::sync::{Arc, Mutex};
+
+use near_async::messaging::{IntoMultiSender, IntoSender, Sender};
+use near_async::test_loop::data::TestLoopDataHandle;
+use near_async::test_loop::sender::TestLoopSender;
+use near_async::time::Duration;
+use near_chain_configs::{ClientConfig, Genesis};
+use near_chunks::shards_manager_actor::ShardsManagerActor;
+use near_client::client_actor::ClientActorInner;
+use near_client::{PartialWitnessActor, ViewClientActorInner};
+use near_jsonrpc::ViewClientSenderForRpc;
+use near_network::shards_manager::ShardsManagerRequestFromNetwork;
+use near_network::state_witness::PartialWitnessSenderForNetwork;
+use near_network::test_loop::{
+    ClientSenderForTestLoopNetwork, TestLoopNetworkSharedState, TestLoopPeerManagerActor,
+    ViewClientSenderForTestLoopNetwork,
+};
+use near_parameters::RuntimeConfigStore;
+use near_primitives::epoch_manager::EpochConfigStore;
+use near_primitives::network::PeerId;
+use near_primitives::types::AccountId;
+use near_primitives::upgrade_schedule::ProtocolUpgradeVotingSchedule;
+use near_store::Store;
+use nearcore::state_sync::StateSyncDumper;
+use tempfile::TempDir;
+
+use super::builder::DropConditionKind;
+use super::env::TestLoopChunksStorage;
+
+const NETWORK_DELAY: Duration = Duration::milliseconds(10);
+
+/// This is the state associate with the test loop environment.
+pub struct SharedState {
+    pub genesis: Genesis,
+    pub tempdir: TempDir,
+    pub epoch_config_store: EpochConfigStore,
+    pub runtime_config_store: Option<RuntimeConfigStore>,
+    pub network_shared_state: TestLoopNetworkSharedState,
+    pub upgrade_schedule: ProtocolUpgradeVotingSchedule,
+    pub chunks_storage: Arc<Mutex<TestLoopChunksStorage>>,
+    pub drop_condition_kinds: Vec<DropConditionKind>,
+    pub load_memtries_for_tracked_shards: bool,
+    pub warmup_pending: Arc<AtomicBool>,
+}
+
+pub struct NodeState {
```

Comment:
may be worth to explicitly differentiate between SharedState, NodeState, and TestData.

Reply:
Added better description for each field and the struct itself.